exercises/queen-attack/queen_attack.example

```diff
@@ -0,0 +1,12 @@
+create((X,Y)) :- X >= 0, Y >= 0, 7 >= X, 7 >= Y.
+
+column((_, Y), (_, V)) :- Y =:= V.
+
+row((X, _), (U, _)):- X =:= U.
+
+diagonal((X, Y), (U, V)) :-  (X-Y) =:= (U-V), !; (X+Y) =:= (U+V).
+
+attack((X, Y), (U, V)) :-
+  column((X, Y), (U, V)), !;
+  row((X, Y), (U, V)), !;
+  diagonal((X, Y), (U, V)).
```

Comment:
Kindof a tangent, am I being overly obfuscating with the following:

```prolog
% Is this overly obtuse? Is there a better way to do this?
% I didn't like how much was repeated, so I basically recreated clojure's juxt here...
attack_map(A, B, Goal) :- call(Goal, A, B).

% Takes in two queens (X,Y) and applies column/row/diagonal to them and keeps
% the Goals that return true. Then, it compares that returned list with [] and
% negates the result.
attack(A, B) :-
  \+ include(attack_map(A, B), [column, row, diagonal], []).
```

It makes me wonder if I'm doing things the hard way due to too much ML/Lisp brain damage.

Reply:
My version it is more explicit and easier to read, I think that for something this simple, it is better to keep the implementation simple too.